### ISSUE TITLE
System.Resources.Extensions API to add untyped BinaryFormatter

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/corefx#37942]
`PreserializedResourceWriter` currently exposes

https://github.com/dotnet/corefx/blob/bf7dbd6d37285a147dd30f5dfca6582046dcabc7/src/System.Resources.Extensions/ref/System.Resources.Extensions.cs#L32

Unfortunately, when (desktop) `ResXResourceWriter` chooses to write a BinaryFormatter byte array, it does not preserve type information in the `.resx`. For instance:

```xml
  <data name="listView1.Groups1" mimetype="application/x-microsoft.net.object.binary.base64">
    <value>
        AAEAAAD/////AQAAAAAAAAAMAgAAAFdTeXN0ZW0uV2luZG93cy5Gb3JtcywgVmVyc2lvbj00LjAuMC4w
        LCBDdWx0dXJlPW5ldXRyYWwsIFB1YmxpY0tleVRva2VuPWI3N2E1YzU2MTkzNGUwODkFAQAAACJTeXN0
        ZW0uV2luZG93cy5Gb3Jtcy5MaXN0Vmlld0dyb3VwBAAAAAZIZWFkZXIPSGVhZGVyQWxpZ25tZW50A1Rh
        ZwROYW1lAQQCAShTeXN0ZW0uV2luZG93cy5Gb3Jtcy5Ib3Jpem9udGFsQWxpZ25tZW50AgAAAAIAAAAG
        AwAAAA1MaXN0Vmlld0dyb3VwBfz///8oU3lzdGVtLldpbmRvd3MuRm9ybXMuSG9yaXpvbnRhbEFsaWdu
        bWVudAEAAAAHdmFsdWVfXwAIAgAAAAAAAAAKBgUAAAAObGlzdFZpZXdHcm91cDIL
</value>
  </data>
```

is accessed at runtime as a `System.Windows.Forms.ListViewGroup`.

In order to pass that resource through, something along the lines of

```c#
public void AddBinaryFormattedResource(string name, byte[] value) { }
```

would be ideal.

/cc @ericstj @dreddy-work

**Comment by ericstj** [comment]
This is required for the WinForms resources work.

**Comment by ericstj** [comment]
Fixed in #38012
